sympy/polys/eigentools.py

```diff
@@ -0,0 +1,115 @@
+from sympy.polys import Poly
+from sympy.polys.domainmatrix import DomainMatrix, DDM, ddm_irref, ddm_isub
+from sympy.polys.factortools import dup_factor_list
+from sympy.polys.agca.extensions import FiniteExtension, ExtensionElement
+from sympy.core.symbol import Dummy
+from sympy.polys.rootoftools import CRootOf
+from sympy.polys.polyroots import roots
+
+
+def ddm_nullspace(M):
+    rows, cols = M.shape
+    domain = M.domain
+
+    pivots = ddm_irref(M)
+    rref = M
+
+    basis = []
+    for i in range(cols):
+        if i in pivots:
+            continue
+        vec = [domain.one if i == j else domain.zero for j in range(cols)]
+        for ii, jj in enumerate(pivots):
+            vec[jj] -= rref[ii][i]
+        basis.append(vec)
+
+    return basis
+
+
+def ddm_from_sympy_matrix(A):
+    rows, cols = int(A.rows), int(A.cols)
+    domain, items = DomainMatrix.get_domain(list(A), field=True, extension=True)
```

Comment:
Take any or all of #20483 in this PR if you want. I added a `DomainMatrix.from_Matrix` method there. It should probably take arguments like field and extension as should `DomainMatrix.from_list_sympy`.

Reply:
Currently, I have no plan to use DomainMatrix for this until the issues of field RREF is resolved for this.

Reply:
Both DDM and DomainMatrix have exactly the same RREF implementation so I'm not sure what that means.

I just mean that it makes more sense for the conversion from Matrix to be handled at the DomainMatrix level rather than DDM. With `from_Matrix()` this function is just `DomainMatrix.from_Matrix(...).rep`.

Reply:
Okay, I can modify the DomainMatrix.from_Matrix to have more options supported. 
But this implementation is temporary  and we need to jump into deciding which options should be valid for from_Matrix or from_list_sympy.

I think that the best option is to wrap all the options for PolyOptions, but I’m not sure if there should be some more options for matrix.

And besides, I think that all the actual computation is done inside DDM so actual computation of eigenvects, nullspace should directly work with DDM since DomainMatrix doesn’t support any indexing.

Reply:
I think DomainMatrix should support indexing in a matrix index/slice sense like `M[:, ::-1]` or `M[i, j]`. I don't think it should have support for all of the complexity that Matrix has including fancy indexing. I also don't think that single indexing is needed so `M[0]` should raise.

We could wrap all the options in polyoptions but for now we could also just support the options that we consider important. As long as they get passed to `construct_domain` they will be wrapped in `PolyOptions` there anyway.

> And this makes it ambiguous whether having methods like rref, inv are meaningful for DomainMatrix.

I don't understand what this means.

Reply:
I notice that lower level methods for DDM uses in-place operations and DomainMatrix can always copy before methods like `rref`, so there can be difference for that.